extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java

```diff
@@ -137,13 +145,18 @@ public void build(NarayanaJtaRecorder recorder,
         builder.addBeanClass(TransactionalInterceptorNotSupported.class);
         additionalBeans.produce(builder.build());
 
+        if (transactionManagerBuildTimeConfig.allowUnsafeMultipleLastResources) {
+            recorder.logAllowUnsafeMultipleLastResources();
```

Comment:
This probably won't fly in 3.8 where we need `allowUnsafeMultipleLastResources` to default to `true` (right?). With this code we'd be displaying a warning on startup for basically every single Quarkus application out there.

I think that in 3.8, when using the default value of the setting (`Optional.empty()`, not `true`), we would need to display the warning on the first occurrence of the problem at runtime (like Narayana) but at most once (unlike Narayana), and probably with a Quarkus-specific message (mentioning the Quarkus-specific configuration property).

Or, simpler to implement: by default we would display a warning *every* occurrence of the problem at runtime, and we'd hide the per-occurrence warning only if `allowUnsafeMultipleLastResources` is explicitly set to `true`... But maybe the risk of flooding the logs is too high, even if setting `allowUnsafeMultipleLastResources` to `true` removes the warning?

Reply:
@aloubyansky @maxandersen Is there a deadline for backporting this to 3.8? Any release planned very soon?